lib/diaspora/exporter/posts_with_activity.rb

```diff
@@ -0,0 +1,68 @@
+module Diaspora
+  class Exporter
+    # This class allows to query posts where a person made any activity (submitted comments,
+    # likes, participations or poll participations).
+    class PostsWithActivity
+      # TODO: docs
+      def initialize(person_id)
+        @person_id = person_id
+      end
+
+      # TODO: docs
+      def query
+        posts_with_activity
+      end
+
+      private
+
+      attr_reader :person_id
+
+      def posts_with_activity
+        Post.from("(#{sql_union_all_activities}) AS posts")
+      end
+
+      def sql_union_all_activities
+        all_activities.map(&:to_sql).join(" UNION ")
+      end
+
+      def all_activities
+        [comments_activity, likes_activity, participation_activity, polls_activity]
+      end
+
+      def likes_activity
+        relayable_activity(like_subquery)
+      end
+
+      def comments_activity
+        relayable_activity(comment_subquery)
+      end
+
+      def participation_activity
+        relayable_activity(participation_subquery)
+      end
+
+      def polls_activity
+        StatusMessage.where.not(author_id: person_id).joins(:poll_participations)
+                     .where("poll_participations.author_id = #{person_id}")
+      end
+
+      def relayable_activity(subquery)
+        Post.where.not(author_id: person_id).joins(subquery)
```

Comment:
Hmm, since we only need the combined query here, I think we should keep it here (and combine it like in `all_activities`), and use the existing scopes from the Post model.